### PR TITLE
add HTTP datasource

### PR DESCRIPTION
**Pull request by he2ss**
feature : https://github.com/crowdsecurity/crowdsec/issues/3289

### CODE REVIEW

File: pkg/acquisition/modules/http/http.go [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+		}
+		tlsConfig.Certificates = []tls.Certificate{cert}
+	}
+
+	if hc.AuthType == "mtls" && hc.TLS.CaCert != "" {
+		caCert, err := os.ReadFile(hc.TLS.CaCert)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read ca cert: %w", err)
+		}
+		caCertPool := x509.NewCertPool()
```
Comment (LaurenceJJones) [inline pkg/acquisition/modules/http/http.go:222]: Some users may want to load from system pool?

ref: https://github.com/crowdsecurity/go-cs-bouncer/blob/9e16e182103d8f9ac30352bff0c1eca27b0619ba/tls.go#L11-L35

File: pkg/acquisition/modules/http/http.go [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+	}
+
+	if hc.Path == "" {
+		hc.Path = "/"
+	}
+	if hc.Path[0] != '/' {
+		return errors.New("path must start with /")
+	}
+
+	switch hc.AuthType {
```
Comment (LaurenceJJones) [inline pkg/acquisition/modules/http/http.go:105]: The errors returned here do not provide context as to why `X is required`

I would expect it to say "Basic auth is selected but no X is provided"

This is small nit pick but could provide enough error context to user to self resolve

**Review by mmetc — COMMENTED** [review]
All data sources should be optionally excluded with build tags: grep for 'datasource_kafka' for an example. They also register with an init() method so you can see with "cscli version" which ones were included

**Review by blotus — APPROVED** [review]

**Pull request merged**